### ISSUE TITLE
Clean up old HGCal reconstruction modules

### ISSUE THREAD

**Issue by kpedro88** [cms-sw/cmssw#31687]
After #31527, the old `hgcalMultiClusters` no longer appear to be used in any sequence. The module and collection should be removed from the default Phase 2 reconstruction, to reduce CPU, memory, disk usage, etc. Any other modules that are no longer needed should also be removed. (If the associated algorithms are not foreseen to be used even in private studies, the source code can be remove as well.)

attn: @rovere @felicepantaleo @cseez

**Comment by kpedro88** [comment]
assign reconstruction,upgrade

**Comment by srimanob** [comment]
Hi
@rovere @felicepantaleo @cseez 
Should this ticket be moved forward? Could it be a target during 12_0 pre-release?

**Comment by jpata** [comment]
type hgcal

**Comment by jpata** [comment]
+reconstruction
- In the latest Phase2 profiles, hgcalMultiClusters does not appear, so I assume it has been addressed (although I could not find the PR now)

**Comment by rovere** [comment]
https://github.com/cms-sw/cmssw/pull/31527 moved the downstream reconstruction to use `TICL`.
I'll review with a little more care if additional cleaning is needed/desired.
